[Bug]: Unexpected error output parsing my package

### Expected Behavior

A more detailed message about the detected problem, if any.

### Actual Behavior

```
Error: Error: Command failed: npm view __ngcc_entry_points__.json@undefined engines --json
npm ERR! code EINVALIDPACKAGENAME
npm ERR! Invalid package name "__ngcc_entry_points__.json" of package "__ngcc_entry_points__.json@undefined": name cannot start with an underscore.
{
  "error": {
    "code": "EINVALIDPACKAGENAME",
    "summary": "Invalid package name \"__ngcc_entry_points__.json\" of package \"__ngcc_entry_points__.json@undefined\": name cannot start with an underscore.",
    "detail": ""
  }
}
```

### Steps to Reproduce

1. execute `depngn 16.0.0` in the Windows command prompt
3.
4.


### Additional Information

_No response_

### What package manager are you seeing the problem with?

npm

### Relevant log output

```shell
0 verbose cli C:\Program Files\nodejs\node.exe C:\Program Files\nodejs\node_modules\npm\bin\npm-cli.js
1 info using npm@8.19.3
2 info using node@v18.13.0
3 timing npm:load:whichnode Completed in 1ms
4 timing config:load:defaults Completed in 2ms
5 timing config:load:file:C:\Program Files\nodejs\node_modules\npm\npmrc Completed in 11ms
6 timing config:load:builtin Completed in 12ms
7 timing config:load:cli Completed in 3ms
8 timing config:load:env Completed in 1ms
9 timing config:load:file:C:\Users\kkopruch\git-repositories\materna\ips\lh-once\ui\ui\.npmrc Completed in 2ms
10 timing config:load:project Completed in 11ms
11 timing config:load:file:C:\Users\kkopruch\.npmrc Completed in 7ms
12 timing config:load:user Completed in 8ms
13 timing config:load:file:C:\Users\kkopruch\AppData\Roaming\npm\etc\npmrc Completed in 3ms
14 timing config:load:global Completed in 3ms
15 timing config:load:validate Completed in 3ms
16 timing config:load:credentials Completed in 1ms
17 timing config:load:setEnvs Completed in 2ms
18 timing config:load Completed in 46ms
19 timing npm:load:configload Completed in 46ms
20 timing npm:load:mkdirpcache Completed in 2ms
21 timing npm:load:mkdirplogs Completed in 2ms
22 verbose title npm view rxjs@7.8.0 engines
23 verbose argv "view" "rxjs@7.8.0" "engines" "--json"
24 timing npm:load:setTitle Completed in 2ms
25 timing config:load:flatten Completed in 6ms
26 timing npm:load:display Completed in 7ms
27 verbose logfile logs-max:10 dir:C:\Users\kkopruch\AppData\Local\npm-cache\_logs
28 verbose logfile C:\Users\kkopruch\AppData\Local\npm-cache\_logs\2023-02-10T07_18_57_889Z-debug-0.log
29 timing npm:load:logFile Completed in 8ms
30 timing npm:load:timers Completed in 0ms
31 timing npm:load:configScope Completed in 0ms
32 timing npm:load Completed in 68ms
33 silly logfile start cleaning logs, removing 11 files
34 silly logfile error removing log file C:/Users/kkopruch/AppData/Local/npm-cache/_logs/2023-02-10T07_18_56_317Z-debug-0.log [Error: EPERM: operation not permitted, unlink 'C:\Users\kkopruch\AppData\Local\npm-cache\_logs\2023-02-10T07_18_56_317Z-debug-0.log'] {
34 silly logfile   errno: -4048,
34 silly logfile   code: 'EPERM',
34 silly logfile   syscall: 'unlink',
34 silly logfile   path: 'C:\\Users\\kkopruch\\AppData\\Local\\npm-cache\\_logs\\2023-02-10T07_18_56_317Z-debug-0.log'
34 silly logfile }
35 silly logfile done cleaning log files
36 http fetch GET 200 https://artifact.materna.de/nexus/repository/ips_mobility_npm_cache/rxjs 209ms (cache revalidated)
37 timing command:view Completed in 232ms
38 verbose exit 0
39 timing npm Completed in 319ms
40 info ok
```


### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/src/queries/getDependencies.ts
+++ b/src/queries/getDependencies.ts
@@ -1,10 +1,23 @@
 import { asyncExec } from './exec';
 import { PackageList, Manager } from '../types';
 
+const AUTO_EXCLUDE = [
+  // automatically added to `node_modules` in Angular projects
+  '__ngcc_entry_points__.json',
+];
+
 export async function getDependencies(manager: Manager): Promise<PackageList> {
   try {
     const list = await asyncExec(manager.list);
-    return JSON.parse(list.stdout).dependencies;
+    const parsedList = JSON.parse(list.stdout).dependencies;
+    return Object.keys(parsedList)
+    .filter((dep) => !AUTO_EXCLUDE.includes(dep))
+    .reduce((acc, curr) => {
+      return {
+        ...acc,
+        [curr]: parsedList[curr],
+      }
+    }, {});
   } catch (error) {
     throw error;
   }

--- a/src/queries/getDependencies.ts
+++ b/src/queries/getDependencies.ts
@@ -11,8 +11,10 @@ export async function getDependencies(manager: Manager): Promise<PackageList> {
     const list = await asyncExec(manager.list);
     const parsedList = JSON.parse(list.stdout).dependencies;
     return Object.keys(parsedList)
-    .filter((dep) => !AUTO_EXCLUDE.includes(dep))
     .reduce((acc, curr) => {
+      if (AUTO_EXCLUDE.includes(curr)) {
+        return acc;
+      }
       return {
         ...acc,
         [curr]: parsedList[curr],

--- a/src/types.ts
+++ b/src/types.ts
@@ -11,7 +11,7 @@ export interface CompatData {
 }
 
 export interface PackageList {
-  [key: string]: PackageData;
+  [depName: string]: PackageData;
 }
 
 export interface PackageData {

--- a/src/types.ts
+++ b/src/types.ts
@@ -10,7 +10,9 @@ export interface CompatData {
   range: string;
 }
 
-export type PackageList = Record<string, PackageData>;
+export interface PackageList {
+  [key: string]: PackageData;
+}
 
 export interface PackageData {
   version: string;

